cmd/query/app/server.go

```diff
@@ -110,6 +110,9 @@ func (s Server) HealthCheckStatus() chan healthcheck.Status {
 func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.MetricsQueryService, options *QueryOptions, tm *tenancy.Manager, logger *zap.Logger, tracer opentracing.Tracer) (*grpc.Server, error) {
 	var grpcOpts []grpc.ServerOption
 
+	if options.GRPCMaxReceiveMessageSize > 0 {
+		grpcOpts = append(grpcOpts, grpc.MaxRecvMsgSize(options.GRPCMaxReceiveMessageSize))
```

Comment:
I don't think this is the right place. This is a [setting for gRPC server][1] that runs as part of jaeger-query. That server allows people to download traces, but it does not _receive_ any large payloads.

The original ticket mentioned gRPC plugin, i.e. where the path is `client <- jaeger-query/gRPC <- grpc-storage-plugin <- remote storage`, and it's in the last hop that throws the exception because the returned message is too large.

[1]: https://pkg.go.dev/google.golang.org/grpc#MaxRecvMsgSize

Reply:
In the original issue, the last hop uses the HTTP/JSON frontend, so there is no gRPC involved there.

The path was

```mermaid
client -->|HTTP/JSON| "jaeger-query" -->|gRPC| "storage-plugin" --> "blackbox out of question"
```

Reply:
In this workflow the proposed change will have no effect whatsoever, because the server whose setting is being changed is not even in the picture. It would be if the client was calling via gRPC, but even then it would have no effect on plugin-blackbox interaction, which is where the error was raised.

Reply:
@yurishkuro No, the error was raised on the layer between jaeger-query and storage-plugin. We are not even using any grpc behind the storage plugin and blackbox, and we aren't even exposing the grpc query port. I have verified that this patch has fixed the issue, at least for the case I reproduced.

Reply:
Well, the error came from query-plugin interaction not plugin-blackbox interaction. But you're right, the changed code affects client-query instead of query-plugin. I wonder how this somehow fixed the issue in my test...